cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/NewTwoWindingsTransformerConversion.java

```diff
@@ -0,0 +1,426 @@
+        AllShunt interpretedShunt = shuntAlternative(cgmesT2xModel, alternative);
+
+        AllPhaseAngleClock interpretedClock = phaseAngleClockAlternative(cgmesT2xModel, alternative);
+        boolean ratio0AtEnd2 = ratio0Alternative(cgmesT2xModel, alternative);
+
+        InterpretedT2xModel interpretedT2xModel = new InterpretedT2xModel();
+        interpretedT2xModel.r = cgmesT2xModel.r;
+        interpretedT2xModel.x = cgmesT2xModel.x;
+
+        interpretedT2xModel.end1.g = interpretedShunt.g1;
+        interpretedT2xModel.end1.b = interpretedShunt.b1;
+        interpretedT2xModel.end1.ratioTapChanger = interpretedTapChanger.ratioTapChanger1;
+        interpretedT2xModel.end1.phaseTapChanger = interpretedTapChanger.phaseTapChanger1;
+        interpretedT2xModel.end1.ratedU = cgmesT2xModel.end1.ratedU;
+        interpretedT2xModel.end1.terminal = cgmesT2xModel.end1.terminal;
+        interpretedT2xModel.end1.phaseAngleClock = interpretedClock.phaseAngleClock1;
+
+        interpretedT2xModel.end2.g = interpretedShunt.g2;
+        interpretedT2xModel.end2.b = interpretedShunt.b2;
+        interpretedT2xModel.end2.ratioTapChanger = interpretedTapChanger.ratioTapChanger2;
+        interpretedT2xModel.end2.phaseTapChanger = interpretedTapChanger.phaseTapChanger2;
+        interpretedT2xModel.end2.ratedU = cgmesT2xModel.end2.ratedU;
+        interpretedT2xModel.end2.terminal = cgmesT2xModel.end2.terminal;
+        interpretedT2xModel.end2.phaseAngleClock = interpretedClock.phaseAngleClock2;
+
+        interpretedT2xModel.ratio0AtEnd2 = ratio0AtEnd2;
+
+        return interpretedT2xModel;
+    }
+
+    private AllTapChanger ratioPhaseAlternative(CgmesT2xModel cgmesT2xModel, Conversion.Config alternative) {
+        TapChangerConversion ratioTapChanger1 = null;
+        TapChangerConversion phaseTapChanger1 = null;
+        TapChangerConversion ratioTapChanger2 = null;
+        TapChangerConversion phaseTapChanger2 = null;
+
+        if (alternative.isXfmr2RatioPhaseEnd1()) {
+            ratioTapChanger1 = combineTapChangers(cgmesT2xModel.end1.ratioTapChanger, cgmesT2xModel.end2.ratioTapChanger);
+            phaseTapChanger1 = combineTapChangers(cgmesT2xModel.end1.phaseTapChanger, cgmesT2xModel.end2.phaseTapChanger);
+        } else if (alternative.isXfmr2RatioPhaseEnd2()) {
+            ratioTapChanger2 = combineTapChangers(cgmesT2xModel.end2.ratioTapChanger, cgmesT2xModel.end1.ratioTapChanger);
+            phaseTapChanger2 = combineTapChangers(cgmesT2xModel.end2.phaseTapChanger, cgmesT2xModel.end1.phaseTapChanger);
+        } else if (alternative.isXfmr2RatioPhaseEnd1End2()) {
+            ratioTapChanger1 = cgmesT2xModel.end1.ratioTapChanger;
+            phaseTapChanger1 = cgmesT2xModel.end1.phaseTapChanger;
+            ratioTapChanger2 = cgmesT2xModel.end2.ratioTapChanger;
+            phaseTapChanger2 = cgmesT2xModel.end2.phaseTapChanger;
+        } else {
```

Comment:
Comment what is the interpretation in this last case

Reply:
This case is now auto-documented after moving interpretation options to enum lists